kuma/wiki/models.py

```diff
@@ -925,6 +929,10 @@ def save(self, *args, **kwargs):
         # Delete any cached last-modified timestamp.
         self.fill_last_modified_cache()
 
+        # Set the shortened URL (e.g. bit.ly).
+        # TODO: Should we do this on a subset of documents? E.g. not redirects?
+        tasks.set_share_url.delay(self.pk)
```

Comment:
True. It wouldn't get a new share URL. Bitly is smart enough to return the same one. But you are right, we can make this smarter I believe. It's not as simple as not setting if it one already exists b/c we need to handle the case of tree moving, where docs may have a `share_url` already but the URL is changing so it needs to be updated.